Source/WebKit/Shared/Cocoa/CoreIPCNSURLRequest.serialization.in

```diff
@@ -0,0 +1,100 @@
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#if PLATFORM(COCOA)
+
+[WebKitSecureCodingClass=NSURLRequest.class] webkit_secure_coding NSURLRequest {
+# obj0: isMutable
+    obj0: CFBooleanRef
+# obj1: URL for the request
+    obj1: URL
+# obj2: Timeout
+    obj2: Number
+# obj3: Cache Policy
+    obj3: Number
+# obj4: Main Document URL
+    obj4: URL?
+# obj5: Should handle HTTP Cookies
+    obj5: CFBooleanRef
+# obj6: Explicit flags
+    obj6: Number
+
+# obj9: Allow cellular
+    obj9: CFBooleanRef
+# obj10: Prevent idle sleep
+    obj10: CFBooleanRef
+# obj11: Expected workload
+    obj11: Number
+# obj12: Time window delay
+    obj12: Number
+# obj13: Time window duration
+    obj13: Number
+# obj14: Network service type
+    obj14: Number
+# obj15: Request Priority
+    obj15: Number
+
+# obj17: Cookie policy
+    obj17: Number?
+# obj18: Request Method
+    obj18: String?
+# obj19: Headers, Dictionary? of <String, Array<String>> | <String, String>
+    obj19: Dictionary?
+# obj20: Request body data for sending
+    obj20: Data?
+# obj21: Body parts, Array<Data|String>?
+    obj21: Array?
+
+    allowPrivateAccessTokensForThirdParty: CFBooleanRef
+    allowedProtocolTypes: Number
+    allowsConstrainedNetworkAccess: CFBooleanRef?
+    allowsExpensiveNetworkAccess: CFBooleanRef?
+    allowsPersistentDNS: CFBooleanRef
+    allowsUCA: CFBooleanRef?
+    assumesHTTP3Capable: CFBooleanRef
+    attribution: Number
+    blockTrackers: CFBooleanRef
+    boundInterfaceIdentifier: String?
+# contentDispositionEncodingFallbackArray: NSArray<NSNumber>
+    contentDispositionEncodingFallbackArray: Array?
+    failInsecureLoadWithHTTPSDNSRecord: CFBooleanRef
+    ignoreHSTS: CFBooleanRef
```

Comment:
I think this is only set in the network process and only used in the network process, so it doesn't need to go across IPC.